exonum-java-proofs/src/test/java/com/exonum/binding/storage/proofs/map/DbKeyComparisonParameterizedTest.java

```diff
@@ -0,0 +1,68 @@
+package com.exonum.binding.storage.proofs.map;
+
+import static com.exonum.binding.test.Bytes.bytes;
+import static com.exonum.binding.test.Bytes.createPrefixed;
+import static com.exonum.binding.test.TestParameters.parameters;
+import static org.hamcrest.CoreMatchers.equalTo;
+import static org.junit.Assert.assertThat;
+
+import java.util.Arrays;
+import java.util.Collection;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.Parameterized;
+import org.junit.runners.Parameterized.Parameter;
+import org.junit.runners.Parameterized.Parameters;
+
+@RunWith(Parameterized.class)
+public class DbKeyComparisonParameterizedTest {
+
+  @Parameter(0)
+  public DbKey firstKey;
+
+  @Parameter(1)
+  public DbKey secondKey;
+
+  @Parameter(2)
+  public boolean expectedResult;
+
+  @Parameter(3)
+  public String description;
+
+  @Test
+  public void commonPrefixTest() {
+    int comparisonResult = firstKey.compareTo(secondKey);
+    assertThat(comparisonResult > 0, equalTo(expectedResult));
+  }
+
+  @Parameters(name = "{index} = {3}")
+  public static Collection<Object[]> testData() {
+    return Arrays.asList(
+        // "A | B -> C" reads "C is a common prefix of A and B"
+        parameters(
+            DbKey.newBranchKey(createUserKey(0b1100), 4),
```

Comment:
it would be great to migrate these tests to the new method `com.exonum.binding.storage.proofs.map.DbKeyCommonPrefixParameterizedTest.branchKeyFromPrefix`

Reply:
We may actually gather the various factory method used in various tests in a utility class under /test directory (e.g., DbKeys#branchKeyFromPrefix, DbKeys#leafKeyFromPrefix, + other)